### PR TITLE
Use Enzyme to test ContributionList dynamics

### PR DESCRIPTION
**Pull request by wchargin**
Summary:
This is our first dynamic test of a React component! Enzyme looks pretty
easy to use to me, for both snapshot tests and interaction simulation.

In doing so, we catch a minor bug in the edge case where a contribution
is not owned by any plugin (`colSpan`, not `colspan`). This edge case
does not appear in the sample data, but it does appear in the test data,
even prior to this commit. The previous renderer, `react-test-renderer`,
appears not to surface this error. Furthermore, this bug did not cause
any user-visible errors except a `console.error`.

Test Plan:
Inspect the snapshot file to make sure that it is reasonable. (The
existing test case has its snapshot regenerated due to formatting
differences between the two renderers.)

To test that the browser error is fixed, render a contribution list on a
GitHub graph but with an empty adapter set. One way to do this is to comment out line 7 of
`standardAdapterSet.js`; alternately, you can use the React Dev Tools to
select the `ContributionList` node, then run
```js
$r.props.adapters.adapters = {};
$r.forceUpdate();
```
Note subsequently that there is no console error and that the `<td>`s in
question span three columns.

wchargin-branch: contributionlist-dynamic-test

### CODE REVIEW

File: src/plugins/artifact/editor/ContributionList.test.js [hunk 1]
```diff
@@ -152,11 +154,47 @@ function createTestData(): * {
...
+      .filterWhere((x) => pattern.test(x.text()));
+    expect(targetOption).toHaveLength(1);
+    container
+      .find("select")
+      .simulate("change", {target: {value: targetOption.prop("value")}});
+  }
+
+  it("renders some test data in the default state", () => {
+    const result = render();
+    expect(result.find("tbody tr")).toHaveLength(5);
```
Comment (dandelionmane) [inline src/plugins/artifact/editor/ContributionList.test.js]: why 5? could this be rewritten as testNodes.length or such?

Comment (wchargin) [reply src/plugins/artifact/editor/ContributionList.test.js]: Could be `createTestData().graph().getAllNodes().length`, although the other case of this assertion would have to be either still `(2)` or something weird that recreates the filter function (i.e., not useful).

Really, though, I might as well just remove these assertions: they're redundant with the snapshot test.

**Comment by wchargin** [comment]
> Can we yarn remove react-test-renderer?

It's still used in `githubPluginAdapter.test.js`. We could potentially replace that with Enzyme, too, though [Jest recommends using `react-test-renderer` for basic snapshot tests](https://facebook.github.io/jest/docs/en/tutorial-react.html#snapshot-testing) and [enzyme for interactions.](https://facebook.github.io/jest/docs/en/tutorial-react.html#dom-testing). I switched the static snapshot test in this file to `enzyme` because I thought it was clearer to have the two snapshots in this file be of the same format.

Any preferences—in general, and for this PR specifically?

File: src/plugins/artifact/editor/ContributionList.test.js (hunk 1 above)
Comment (wchargin) [reply src/plugins/artifact/editor/ContributionList.test.js]: > Really, though, I might as well just remove these assertions: they're redundant with the snapshot test.

Although, after removing them, we should keep in mind that if the snapshots undergo a ton of churn, then we could easily replace them with more direct tests like this.

**Comment by dandelionmane** [comment]
Overall, my bias is to have fewer tools, and having two tools that essentially do the same thing (but one is more general-purpose than the other) is unappetizing. Looking at the docs you linked, it sounds like there might be an issue around mocking if we use react-test-renderer.

I think my vote would be: remove react-test-renderer, with a strong willingness to add it back if/when it turns out we really need it. But that leaves us with the bias towards simpler tooling, and adding complexity only at the point where it is clearly providing additional value. Fine with doing that in a separate PR.

**Comment by dandelionmane** [comment]
Also relevant (arguably fixes): https://github.com/sourcecred/sourcecred/issues/25

**Comment by wchargin** [comment]
> I think my vote would be: remove react-test-renderer, with a strong willingness to add it back if/when it turns out we really need it. But that leaves us with the bias towards simpler tooling, and adding complexity only at the point where it is clearly providing additional value. Fine with doing that in a separate PR.

Sounds good to me.
